go/service/gregor.go

```diff
@@ -942,6 +944,46 @@ func (g *gregorHandler) chatTlfFinalize(ctx context.Context, m gregor.OutOfBandM
 	return nil
 }
 
+func (g *gregorHandler) chatTlfResolve(ctx context.Context, m gregor.OutOfBandMessage) error {
+	if m.Body() == nil {
+		return errors.New("gregor handler for chat.tlfresolve: nil message body")
+	}
+
+	g.G().Log.Debug("push handler: tlf resolve received")
+
+	var update chat1.TLFResolveUpdate
+	reader := bytes.NewReader(m.Body().Bytes())
+	dec := codec.NewDecoder(reader, &codec.MsgpackHandle{WriteExt: true})
+	err := dec.Decode(&update)
+	if err != nil {
+		return err
+	}
+
+	uid := m.UID().String()
+
+	// Get and localize the conversation to get the new tlfname.
+	inbox, _, err := g.G().InboxSource.ReadNoCache(ctx, m.UID().Bytes(), nil, &chat1.GetInboxLocalQuery{
```

Comment:
Can `InboxSource.Read` be used here to hit the cache? Or does that only work with some queries? Or would it cause the tlfname not to be re-resolved?

Also, is this slow enough or have the potential to hang and if so should put this notifyrouter send in a goroutine?